include/zephyr/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -35,7 +35,7 @@
 #if CONFIG_FLASH_LOAD_SIZE > 0
 #define ROM_SIZE CONFIG_FLASH_LOAD_SIZE
 #else
-#define ROM_SIZE (CONFIG_FLASH_SIZE*1K - CONFIG_FLASH_LOAD_OFFSET)
+#define ROM_SIZE (CONFIG_FLASH_SIZE * 1024 - CONFIG_FLASH_LOAD_OFFSET)
```

Comment:
Is that change required for this PR?

Reply:
yes, because this value is used in preprocessor comparison (`#if ROM_SIZE > ...`) and preprocessor does not understand `1K` while linker does.